### ISSUE TITLE
server exited / failed to write request: Broken pipe (os error 32)

### ISSUE THREAD

**Issue by Zoybean** [rust-lang/rust-analyzer#13998]
RA reports an error on any project using `thiserror`'s derive macro, 

**rust-analyzer version**: `rust-analyzer b594f9c44 2022-03-21 stable`

**rustc version**: `rustc 1.66.1 (90743e729 2023-01-10)`

**relevant settings**: none of the listed environment variables are set. RA configuration unchanged

**minimal project:**
cargo.toml:
```
[package]
name = "min-error"
version = "0.1.0"
edition = "2021"

[dependencies]
thiserror = "1.0.38"
```
src/lib.rs:
```
use thiserror::Error;
#[derive(Error, Debug)]
enum MyError {}
```

using [`helix` editor](https://helix-editor.com/) with RA as the language server

`cargo check` output: no errors or warnings
Expected RA behaviour: no errors or warnings
Actual RA behaviour:
- Upon opening the file, the `#[derive]` and `enum` are marked with the error `server exited`.
- After editing the file at all (e.g. adding a new line at the start) this error is replaced by `failed to write request: Broken pipe (os error 32)`.

**Comment by lnicola** [comment]
Can you try a newer version of RA? Yours is slightly dated.

**Comment by Zoybean** [comment]
I installed it with `rustup` and have run `rustup update` earlier today, but it looks like I have another version too, somehow. That's probably where this is coming from. I'll resolve that and let you know if the issue persists

**Comment by Zoybean** [comment]
Issue was tied to outdated rustup version. Now it's just a mystery how I had that version to begin with.

**Comment by lnicola** [comment]
Yeah, it's usually in ~/.cargo/bin, /usr/local/bin or /usr/bin,. depending on how persistent you were while installing it. The which command should help.

But I suggest downloading a binary from GitHub and keeping it updated. The version included with a stable toolchain will be pretty old (maybe months). Nightlies will be newer (maybe weeks old). You can mix and match.
